### PR TITLE
Use Standard Dependency Workflow

### LINKED ISSUE #289: Implement Dependency RFC 0004 for  pip
Implement [Dependency Management RFC Phase 1](https://github.com/paketo-buildpacks/rfcs/blob/main/text/dependencies/rfcs/0004-dependency-management-phase-one.md) for `pip`. Check out the RFC for more details and background.

### 1. Determine dependency source strategy
<details>
<summary>Background</summary>

When possible, dependencies should be used directly from upstream, rather than undergoing any additional compilation or modifications performed by Paketo-maintained code. For each dependency, the corresponding buildpack maintainer group will decide if the dependency can be used directly from upstream, and must identify the location from which the dependency will be pulled from. Some of the Paketo Java buildpacks perform directory stripping during the buildpack build process itself. This could be a viable alternative to performing directory modifications during the dependency management process for maintainers to consider.
</details>

- [ ] Determine whether the dependency can be used directly from its upstream, rather than undergoing additional compilation/modification. 
* If using directly from upstream, note where dependencies will come from
* If compiled or modified, note the decision for this
- [ ] Document decision in a language-family level RFC. This decision can be combined in the RFC with the decision made for other buildpacks in the language family.

### 2. Version retrieval and metadata generation code
Please refer to the [retrieval RFC section](https://github.com/paketo-buildpacks/rfcs/blob/main/text/dependencies/rfcs/0004-dependency-management-phase-one.md#1-version-retrieval-make-retrieve) on version retrieval for details around the API and background.

- [ ] Create retrieval code for the dependency
* Will live in the buildpack under: `dependency/retrieval/`
* Per [the RFC](https://github.com/paketo-buildpacks/rfcs/blob/main/text/dependencies/rfcs/0004-dependency-management-phase-one.md#transfercombine-version-retrieval-and-metadata-generation), this will be a combination of (1) discovering new dependency versions based on the `buildpack.toml`, and (2) generating metadata for each new version.
* **Note** : Per [the RFC caveat](https://github.com/paketo-buildpacks/rfcs/blob/main/text/dependencies/rfcs/0004-dependency-management-phase-one.md#caveat-compiled-dependencies) - if the dependency is to be compiled, the `SHA256` and `URI` field from the metadata should be omitted in this step.

### 3. Compilation code (if needed)
If the dependency will be compiled/modified, then refer to the [compilation RFC section](https://github.com/paketo-buildpacks/rfcs/blob/main/text/dependencies/rfcs/0004-dependency-management-phase-one.md#2-compilation) for API details and background. 

- [ ] Create compilation action
* Will live in the buildpack under: `dependency/actions/compile/`

### 4. Dependency testing (optional)
It's up to maintainer discretion if the dependency will be tested. It's recommended to test the dependency if it's been compiled. Check out the [testing RFC section](https://github.com/paketo-buildpacks/rfcs/blob/main/text/dependencies/rfcs/0004-dependency-management-phase-one.md#3-test-dependency-make-test) for details.

- [ ] Add tests for dependency
* Will live in the buildpack under: `dependency/test.`

### 5. Makefile Setup
When using the generalized workflows for dependency management down the line, version retrieval and dependency testing will be executed via a Makefile in order to provide the workflow a standardized way to run the code, regardless of what language it was written in. Check out the [RFC section](https://github.com/paketo-buildpacks/rfcs/blob/main/text/dependencies/rfcs/0004-dependency-management-phase-one.md#overarching-plan) for what this should look like.

- [ ] Add Makefile
* Will live in the buildpack at: `dependency/Makefile`

### 6. Leveraging the new code
This issue serves to set up all the main logic for the dependency management. The work to actually leverage this code and migrate off of the dep-server will be completed in a separate issue once workflows and infrastructure is set up.

### PR DESCRIPTION
**Pull request by joshuatcasey**
## Summary
Use the standard dependency workflow. Resolves #289 

Note that the logic in `dependency/actions/compile/entrypoint` seems much simpler than the logic in [`buildpacks-ci`](https://github.com/cloudfoundry/buildpacks-ci/blob/f4cf2db0338254e5c346b7c22d34fe8fc0338cdb/tasks/build-binary-new/builder.rb#L31-L62), yet it seems to achieve the same result. This is verified by running the compilation step for `22.2.2` and comparing the result to https://deps.paketo.io/pip/pip_22.2.2_linux_noarch_bionic_97bb6b2d.tgz.

Note the version of the included `setuptools`. In `buildpacks-ci` this seems to be pinned to `62.1.0`, yet in https://deps.paketo.io/pip/pip_22.2.2_linux_noarch_bionic_97bb6b2d.tgz it's clearly `63.4.1`. Why is that? Can we allow it to just float to the latest?

### CODE REVIEW

File: .github/.syncignore [hunk 1]
```diff
@@ -1,2 +1,2 @@
 CODEOWNERS
-workflows/update-dependencies-from-metadata.yml
+workflows/update-dependencies.yml
```
Comment (robdimsdale) [inline .github/.syncignore:2]: If we delete the `update-dependencies.yml` workflow file, should we also remove this entry from the `syncignore`?

**Review by robdimsdale — APPROVED** [review]
This looks great, with one small question about the `syncignore` file.

**Comment by robdimsdale** [comment]
Thanks for putting this together @joshuatcasey .

I think this simplified logic for `pip` download/install is appropriate. In the past we've had to do more complex things (like pinning `setuptools` to a specific version) due to incompatibilities between versions of `pip` and `python`. We may find we have to do that again, should certain versions of `pip` not work on certain versions of `python`.

File: .github/.syncignore (hunk 1 above)
Comment (joshuatcasey) [reply .github/.syncignore:2]: We should leave it in `.github/.syncignore` so that the `github-config` sync doesn't re-add the file.

**Comment by joshuatcasey** [comment]
@robdimsdale Could you reapprove? I rearranged the commits and rebased on main.

I also removed `linux_64` from the artifact name. The artifact produced is just reorganized source files.

**Review by robdimsdale — APPROVED** [review]

**Pull request merged**